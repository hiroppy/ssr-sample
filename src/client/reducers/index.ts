import { combineReducers } from 'redux';
import {
  reducer as pagesReducer,
  initialState as pagesInitialState,
  State as PagesState
} from './pages';
import {
  reducer as sagaPageReducer,
  initialState as sagaPageInitialState,
  State as SagaPageState
} from './sagaPage';

export type State = {
  pages: PagesState;
  sagaPage: SagaPageState;
};

export const initialState = {
  pages: pagesInitialState,
  sagaPage: sagaPageInitialState
};

export const rootReducer = combineReducers({
  pages: pagesReducer,
  sagaPage: sagaPageReducer
});
