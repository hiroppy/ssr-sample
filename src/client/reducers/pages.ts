import { Actions } from '../actions/pages';

export type State = {
  isLoadingCompletion: boolean;
  error: Error | null;
  baseUrl: string;
  env: typeof process.env.NODE_ENV;
};

export const initialState: State = {
  isLoadingCompletion: false,
  error: null,
  baseUrl: '',
  env: 'development'
};

export const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case 'SET_ENV':
      return { ...state, env: action.payload.env };
    case 'RESET_PAGES_STATUS':
      return initialState;
    case 'SET_BASE_URL':
      return { ...state, baseUrl: action.payload.baseUrl };
    case 'LOAD_TOP_PAGE':
    case 'LOAD_SAGA_PAGE':
    case 'LOAD_APOLLO_PAGE':
      return { ...state, isLoadingCompletion: false };
    case 'LOAD_TOP_PAGE_SUCCESS':
    case 'LOAD_SAGA_PAGE_SUCCESS':
    case 'LOAD_APOLLO_PAGE_SUCCESS':
      return { ...state, isLoadingCompletion: true };
    case 'LOAD_TOP_PAGE_FAILURE':
    case 'LOAD_SAGA_PAGE_FAILURE':
    case 'LOAD_APOLLO_PAGE_FAILURE':
      return { ...state, isLoadingCompletion: true, error: action.payload.err };
    default:
      return state;
  }
};
