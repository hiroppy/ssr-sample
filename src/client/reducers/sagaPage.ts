import { Actions } from '../actions/fetchSaga';
import { Samples } from '../../server/responseSchema';

export type State = {
  samples: Samples;
};

export const initialState: State = {
  samples: []
};

export const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case 'FETCH_SAGA_CODE_SUCCESS':
      return { ...state, samples: action.payload.samples };
    case 'FETCH_SAGA_CODE':
    case 'FETCH_SAGA_CODE_FAILURE':
    default:
      return state;
  }
};
