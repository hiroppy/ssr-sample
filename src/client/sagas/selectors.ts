import { State } from '../reducers';

export const getUsers = (state: State) => state.users;
export const getOrgs = (state: State) => state.orgs;
