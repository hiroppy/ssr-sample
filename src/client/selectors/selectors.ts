import { State } from '../reducers';

export const getEnv = (state: State) => state.pages.env;
export const getBaseUrl = (state: State) => state.pages.baseUrl;
export const getSagaCode = (state: State) => state.sagaPage.samples;
