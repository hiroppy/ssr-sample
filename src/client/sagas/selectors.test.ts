import { createStore } from 'redux';
import * as selectors from './selectors';
import { rootReducer } from '../reducers';

test('should get users state', () => {
  const storeState = createStore(rootReducer).getState();

  expect(selectors.getUsers(storeState as any)).toMatchSnapshot();
});

test('should get orgs state', () => {
  const storeState = createStore(rootReducer).getState();

  expect(selectors.getOrgs(storeState as any)).toMatchSnapshot();
});
