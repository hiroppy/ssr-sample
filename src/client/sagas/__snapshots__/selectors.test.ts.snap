// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get orgs state 1`] = `
Object {
  "isFetching": false,
  "name": "",
  "repos": Array [],
}
`;

exports[`should get users state 1`] = `
Object {
  "name": "",
}
`;
